src/adagrams.js

```diff
@@ -1,15 +1,87 @@
+import { 
+  LETTER_POOL,
+  LETTER_SCORE,
+} from "constants.js";
+
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let letterPoolCopy = {...LETTER_POOL};
+  let hand = [];
+  while (hand.length < 10) {
+    let letter = Object.keys(letterPoolCopy)[Math.floor(Math.random()*Object.keys(letterPoolCopy).length)];
+    hand.push(letter);
+    letterPoolCopy[letter]--;
+    if (letterPoolCopy[letter] === 0) {
+      delete letterPoolCopy[letter];
+    }
+  }
+  return hand;
```

Comment:
Nice logic! A few thoughts:
- Smart idea making a copy of the pool on line 9 to avoid mutating the input.
- All of the variables can be `const`s because we only ever mutate them, not reassign them.
- Consider rewriting line 12 to be multiple statements. There's a lot going on and it's a bit hard to read.